controller/pom.xml

```diff
@@ -168,4 +168,45 @@
         </dependency>
     </dependencies>
 
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>default-test</id>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <configuration>
+                            <environmentVariables>
+                                <!-- Used in EnvVarAttributeOverrideModelTestCase -->
+                                <!-- This corresponds to the /subsystem=custom resource and its my-attr attribute -->
+                                <SUBSYSTEM_CUSTOM__MY_ATTR>value2</SUBSYSTEM_CUSTOM__MY_ATTR>
+                            </environmentVariables>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>env-var-overrides</id>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <configuration>
+                            <environmentVariables>
+                                <!-- Used in EnvVarAttributeOverrideModelTestCase -->
+                                <!-- The presence of this env var is required to trigger the overridding feature -->
+                                <WILDFLY_OVERRIDING_ENV_VARS>1</WILDFLY_OVERRIDING_ENV_VARS>
+                                <!-- This corresponds to the /subsystem=custom resource and its my-attr attribute -->
+                                <SUBSYSTEM_CUSTOM__MY_ATTR>value2</SUBSYSTEM_CUSTOM__MY_ATTR>
+                            </environmentVariables>
+                            <includes>
+                                <include>org/jboss/as/controller/EnvVarAttributeOverrideModelTestCase.java</include>
```

Comment:
Running the same test in two different executions doesn't work well in CI as failures in the 2nd execution don't get reported by TeamCity. Perhaps duplicate the test and exclude the one for this execution from the default execution?

I wonder if changing the name of EnvVarAttributeOverrideModelTestCase to match the pattern that gets tested would work, and then add 2 trivial public static inner classes. So it all stays in 1 .java file.

Reply:
updated